lib/assets/uploader.js

```diff
@@ -0,0 +1,176 @@
+  window.apiVideoFlutterUploader.standardUploaders = window.apiVideoFlutterUploader.standardUploaders || [];
+  window.apiVideoFlutterUploader.standardUploaders.push(uploader);
+}
+
+function createProgressiveUploadWithUploadTokenSession(sessionId, uploadToken, videoId) {
+  return progressiveUploadCreationHelper({
+    sessionId,
+    uploadToken,
+    videoId: videoId || undefined,
+  });
+}
+
+function createProgressiveUploadWithApiKeySession(sessionId, apiKey, videoId) {
+  return progressiveUploadCreationHelper({
+    sessionId,
+    apiKey,
+    videoId: videoId || undefined,
+  });
+}
+
+async function uploadPart(sessionId, filePath, onProgress) {
+  return await uploadPartHelper(sessionId, filePath, onProgress, async (session, blob) => {
+    return await session.uploader.uploadPart(blob);
+  });
+}
+
+async function uploadLastPart(sessionId, filePath, onProgress) {
+  return await uploadPartHelper(sessionId, filePath, onProgress, async (session, blob) => {
+    return await session.uploader.uploadLastPart(blob);
+  });
+}
+
+async function getBlobFromPath(filePath) {
+  return await fetch(filePath)
+    .then(r => r.blob());
+}
+
+async function cancelAll() {
+  const sessions = window.apiVideoFlutterUploader.progressiveSessions;
+  if (sessions != null) {
+    Object.values(sessions).forEach(session => {
+      session.uploader.cancel();
+    });
+  }
+
+  const standardUploaders = window.apiVideoFlutterUploader.standardUploaders;
+  if (standardUploaders != null) {
+    standardUploaders.forEach(uploader => {
+      uploader.cancel();
+    });
+  }
+}
+
+async function jsDisposeProgressiveUploadSession(sessionId) {
+  const session = getProgressiveSession(sessionId);
+  if (session != null) {
+    delete window.apiVideoFlutterUploader.progressiveSessions[sessionId];
+  }
+}
+
+async function uploadWithUploadToken(filePath, uploadToken, videoName, onProgress, videoId) {
+  return uploadHelper(filePath, onProgress, { uploadToken, videoName, videoId });
+}
+
+async function uploadWithApiKey(filePath, apiKey, onProgress, videoId) {
+  return uploadHelper(filePath, onProgress, { apiKey, videoId });
+}
+
+
+function progressiveUploadCreationHelper(options) {
+  const uploader = new ProgressiveUploader({
+    ...options,
+    origin: getOriginHeader(),
+  });
+
+  uploader.onProgress((e) => {
+    const onProgress = getProgressiveSession(options.sessionId).partsOnProgress[e.part];
+    if (onProgress) {
+      onProgress(e.uploadedBytes / e.totalBytes);
+    }
+  });
+
+  setProgressiveSession(options.sessionId, {
+    uploader,
+    partsOnProgress: {},
+    currentPart: 1,
+  });
+
+  return Promise.resolve();
+}
+
+async function uploadPartHelper(sessionId, filePath, onProgress, uploadCallback) {
+  const blob = await getBlobFromPath(filePath);
+  const session = getProgressiveSession(sessionId);
+
+  if (onProgress != null) {
+    session.partsOnProgress[session.currentPart] = onProgress;
+  }
+
+  session.currentPart++;
```

Comment:
I am afraid having 2 `currentPart` (on in the ts uploader, one here) might be a source of bug in the future, but I don't any better idea.